src/protections/MentionSpam.ts

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2024 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Protection } from "./IProtection";
+import { Mjolnir } from "../Mjolnir";
+import { LogLevel, Permalinks, UserID } from "@vector-im/matrix-bot-sdk";
+import { NumberProtectionSetting } from "./ProtectionSettings";
+
+export const DEFAULT_MAX_MENTIONS = 10;
+const USER_ID_REGEX = /@[^:]*:.+/;
```

Comment:
can we just limit this to number of `@` symbols? (plus HTML encoded varient for htmlBody)

Reply:
I think there are concrete chances of that reacting on somebody posting a snippet of code with some delimiter:
```ini
some = value
;@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
next = section
```

Reply:
"don't do that in my rooms" :p